modules/build/src/main/scala/scala/build/bsp/protocol/DiagnosticData.scala

```diff
@@ -0,0 +1,51 @@
+package scala.build.bsp.protocol
+
+import ch.epfl.scala.bsp4j as b
+import com.google.gson.Gson
+import com.google.gson.JsonElement
+
+/** Representation for the data field in a bsp.Diagnostic.
+  *
+  * For now this only contains an edit, but there is nothing from keeping it having more field in
+  * the future.
+  * @param edit
+  *   The Workspace edit attatched to the Diagnostic.
+  */
+case class DiagnosticData(edit: WorkspaceEdit) {
+
+  /** Go from the DiagnosticData to Json. This will result in each param being a field in the
+    * object.
+    *
+    * @return
+    *   the JsonElement
+    */
+  def toJsonTree(): JsonElement = new Gson().toJsonTree(this)
+}
+
+/** A WorkspaceEdit meant to mimic the LSP WorkspaceEdit.
+  *
+  * https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspaceEdit
+  *
+  * For now this only utilizes changes, hench the default empty for both documentChanges and
+  * changeAnnotations. In the future when we support more operations like, CreateFile, RenameFile,
+  * etc we can expland on this and change the type of documentChanges to actually be an
+  * Array[TextDocumentEdit] | Array[TextDocumentEditTypes].
+  *
+  * @param changes
+  *   A Map of the URI that contains the diagnostic to the TextEdits that it contains.
+  * @param documentChanges
+  *   NOT CURRENTLY USED
+  * @param changeAnnotations
+  *   NOT CURRENTLY USED
+  */
+case class WorkspaceEdit(
+  changes: Map[String, Array[TextEdit]],
+  documentChanges: Array[Object] = Array.empty,
+  changeAnnotations: Map[String, Object] = Map.empty
```

Comment:
I wasn't 100% sure what to do with these as I could create an `enum` for the different type of `documentChanges`, but we don't use any of them at the moment. The same with the `changeAnnotations`. So I opted to just use an `Object` here and make them empty. Any thoughts on this or a better alternative? Should I make the entire structure right away even though we aren't using it? During `toJsonTree()` these also get added to the json object with an empty array. I'll need to check if that screws up clients since they might check if `documentChanges` _exists_ and if it does, they might not check `changes`.

Reply:
if we're fairly sure what values we expect there in the future, then defining the entire structure (including enums) straight away sounds good, even if we wouldn't be using them just yet... especially as this will require corresponding changes in clients, if I understand correctly. 
So the earlier the model is there, the better. But I consider it nice-to-have.

Reply:
Alright, so I broke this out and basically made all the `ResourceOperation`s. I did make a few assumptions though about the spec that I'll comment inline.